django/db/migrations/operations/fields.py

```diff
@@ -283,7 +281,7 @@ def state_forwards(self, app_label, state):
                     [self.new_name if n == self.old_name else n for n in together]
                     for together in options[option]
                 ]
-        state.reload_model(app_label, self.model_name_lower, delay=delay)
+        state.reload_model(app_label, self.model_name_lower)
```

Comment:
During [the state reload](https://github.com/django/django/blob/eabd9677261805eb2ef1b5de3188e44852f71961/django/db/migrations/operations/fields.py#L206) at the end of the first migration operation, `state._get_related_models()` returns different results for the migrate and sqlmigrate cases. The code path is the same but ` _get_related_models(<class '__fake__.Member'>)` returns `[<class '__fake__.Agreement'>, <class '__fake__.Group'>]` for `sqlmigrate` and `[<class '__fake__.Group'>]` for `migrate`. It looks like some difference with the model meta relation tree as [m._meta.get_fields()](https://github.com/django/django/blob/2328f2faf8536704662155fc94c06ca0d13e82ac/django/db/migrations/state.py#L35) doesn't return the reverse relation from `Member.id` to `Agreement` in the latter case. As you said on the ticket, without `Agreement` reloaded,  [_related_non_m2m_objects()](https://github.com/django/django/blob/eabd9677261805eb2ef1b5de3188e44852f71961/django/db/backends/base/schema.py#L551) doesn't return related objects.

Let me know if this gives you any ideas. I'll continue if needed but debugging this is proving very time consuming.